applications/launchpad/backend/src/grpc/mod.rs

```diff
@@ -61,13 +64,21 @@ impl TryFrom<TransactionEvent> for WalletTransaction {
     }
 }
 
-impl From<GetIdentityResponse> for WalletIdentity {
-    fn from(value: GetIdentityResponse) -> WalletIdentity {
-        WalletIdentity {
+impl TryFrom<GetIdentityResponse> for WalletIdentity {
+    type Error = String;
+
+    fn try_from(value: GetIdentityResponse) -> Result<Self, Self::Error> {
+        let hex_public_key = String::from_utf8(value.public_key.clone()).unwrap();
+        let emoji_id = EmojiId::from_hex(&hex_public_key)
+            .map_err(|e| format!("Failed to create an emoji: {}", e))?
+            .as_str()
+            .to_string();
```

Comment:
```suggestion
        let emoji_id = EmojiId::from_bytes(&value.public_key)
            .map_err(|e| format!("Could not create emojiId from invalid public key: {}", e))?
            .to_string();
```

Reply:
but EmojiId::from_bytes is not a public fn. Should I change it?

Reply:
Ah, you're right. I had thought `ByteArray` was implemented for EmojiId.